### ISSUE TITLE
Expected Performance with Ideal Pose

### ISSUE THREAD

**Issue by goksanisil23** [rsasaki0109/graphslam_ros2#30]
Hi,

Firstly thanks for providing this package, especially in ROS2. 

My question is about usage of ideal odometry data (and the corresponding tf) for mapping purposes.

The odometry data I'm using is "ideal" in the sense that its from an RTK system with centimeter accuracy. I'm not using IMU data. However when the pose from the SLAM algorithm is updated with the pose from pointcloud matching, I can see a visible drift at the point of loop closure. So the modified_map that's generated at the end has a visible mismatch at the loop closure point (but the input odometry data is continous, without any jump)
Is there some way to increase the weight of the pose information from the odometry, or some constraint parameter?

Thanks and regards

**Comment by rsasaki0109** [comment]
Thank you for your interest in this package. Unfortunately, it is currently not possible to use RTK-GNSS well with this package.

If someone could provide me with sample data, I could develop a RTK-GNSS combined function.

**Comment by boostthrill** [comment]
I could provide you with a ROSBAG with RTK-GNSS ( /fix /heading) and the PointCloud of a Livox MID 70. Would that help you to program a combined function.

**Comment by rsasaki0109** [comment]
@boostthrill 
Yes, the provided ROSBAG data will be very helpful for developing the combined functionality.
